### DISCUSSION TITLE
Parallelize Fortran compilation better by first creating .mod files with --syntax-only

### DISCUSSION THREAD

**Discussion by sujona** [mesonbuild/meson#12963] [Ideas]
In Fortran, the dependencies are in principle such that a file needs the `.mod` files of all `use`d modules. You can run `gfortran` with `--syntax-only` to get the `.mod` files without compiling. This needs to be done in the right order according to the dependencies, but it is very fast. After that I think all the sources can be compiled in parallel without regards to the dependencies. 

If this works, it could speed up the compilation quite a bit. I often see a few dependent files basically compiling in serial in the later stages.

**Comment by dcbaker** [comment]
Would you be interested in testing https://github.com/mesonbuild/meson/pull/12539 or pointing me at the projects you're looking at so I can test them? That *should* allow much greater parallelism by telling Ninja which Fortran sources produce which module interfaces and then ordering each source compilation on it's actual dependencies instead of full target dependencies
